### PR TITLE
[SPARK-9356][SQL]Remove the internal use of DecimalType.Unlimited

### PR DESCRIPTION
**Pull request by yjshen**
JIRA: https://issues.apache.org/jira/browse/SPARK-9356

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala [hunk 1]
```diff
@@ -179,8 +179,7 @@ private[sql] object PartitioningUtils {
    * {{{
    *   NullType ->
    *   IntegerType -> LongType ->
-   *   DoubleType -> DecimalType.Unlimited ->
-   *   StringType
+   *   DoubleType -> StringType
```
Comment (yjshen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala]: internally, it uses
```
private val upCastingOrder: Seq[DataType] =
    Seq(NullType, IntegerType, LongType, FloatType, DoubleType, StringType)
```
to do type upcasting, no decimal type is involved

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala [hunk 2]
```diff
@@ -509,8 +509,6 @@ case class Cast(child: Expression, dataType: DataType)
   private[this] def changePrecision(d: String, decimalType: DecimalType,
       evPrim: String, evNull: String): String = {
     decimalType match {
-      case DecimalType.Unlimited =>
-        s"$evPrim = $d;"
       case DecimalType.Fixed(precision, scale) =>
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala]: you can remove the pattern match entirely. just do decimalType.precision / decimalType.scale

Comment (yjshen) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala]: Get it.

**Comment by yjshen** [comment]
Jenkins, retest this please.

**Comment by rxin** [comment]
Thanks - I've merged this.

**Pull request closed without merging**